### DISCUSSION TITLE
Ros2 Control and spot description overlap

### DISCUSSION THREAD

**Discussion by IoTDan** [bdaiinstitute/spot_ros2#480] [Q&A] [answered]
hello! I see a lot of great work happening in the https://github.com/bdaiinstitute/spot_ros2/tree/main/spot_ros2_control code.
I wanted to understand better if the overlap with https://github.com/bdaiinstitute/spot_ros2/tree/main/spot_description is intentional, specifically in teh XACRO being in both places.

I realize the spot_ros2_control is relatively new.

**Comment by khughes-bdai** [comment]
Hi!
The `spot_description` package is meant to contain the core URDF and meshes for Spot. The `spot_ros2_control` package builds upon the xacro in `spot_description` by adding the necessary tags for ROS 2 control. It would have also been possible to add this into the `spot_description` package instead, but we chose to design it like this to modify as little of the existing packages as possible so that all of the ros2 control code was isolated in this one package. The package is still very new though, and I would be open to moving all of the xacros into `spot_description` if it made things more concise.

**Comment by IoTDan** [comment]
Hello! Thank you very much for getting back to me!

It was super awesome to see the ROS2 control code get checked in!

The scenario I was working on was trying to get the Simulated spot, ROS2 control and NVIDIA Omniverse working.

I respect your POV. Having the XACRO/URDR in one consistent place I think would be beneficial. That way what ends up (including the default included URDF) in Omniverse samples as well as what is pulled from the repo will (hopefully) stay in sync. Adding a 3rd copy in spot_ros2_control only added to confusion.

**Comment by khughes-bdai** [comment]
@IoTDan that makes a lot of sense -- I can put up a PR within the next few days to consolidate everything into one xacro.

**Comment by khughes-bdai** [comment, marked as answer]
Consolidated in https://github.com/bdaiinstitute/spot_ros2/pull/484 --- glad you are interested in the ros2 control additions! It's still a bit of a work in progress, appreciate the feedback :)
